### PR TITLE
Try to improve Compilation Speeds

### PR DESCRIPTION
**Pull request by avik-pal**
- [x] Recompile Invalidations
- [x] Remove Method Ambiguities
- [x] Test using Aqua
- [x] Lift Closures out
    - [x] Single Shooting
    - [x] Multiple Shooting
    - [x] MIRK
- [x] Make Tests Pass
- [x] Precompile Some common workloads
- [x] Upstream Partial Functions Patch https://github.com/SciML/SciMLBase.jl/pull/531
- [x] Remove the Manifest File
    - [x] https://github.com/SciML/NonlinearSolve.jl/pull/265
    - [x] https://github.com/SciML/DiffEqBase.jl/pull/952
    - [x] https://github.com/JuliaDiff/SparseDiffTools.jl/pull/266
- [x] Wait for https://github.com/archermarx/PartialFunctions.jl/issues/8 or just use anonymous functions

### CODE REVIEW

**Comment by avik-pal** [comment]
Seems like MIRK is not causing any trouble. But Shooting Methods precompilation freezes in v1.9

**Comment by avik-pal** [comment]
I am printing a warning about the TTFX for Shooting Methods in v1.9

File: Manifest.toml [hunk 1]
```diff
@@ -0,0 +1,994 @@
+# This file is machine-generated - editing it directly is not advised
+
+julia_version = "1.9.3"
```
Comment (ChrisRackauckas) [inline Manifest.toml:3]: can be removed?

Comment (avik-pal) [reply Manifest.toml:3]: Yeah mostly

**Comment by avik-pal** [comment]
But let me try to use SnoopCompile and check how to bring the timings down

**Comment by avik-pal** [comment]
The multiple shooting code has too many runtime dispatches when tested with JET which is causing the nightmarish precompile times

**Comment by avik-pal** [comment]
Part of it comes from the chunksize not being specified.

**Comment by avik-pal** [comment]
(Now we cross our fingers and wait for the CI to run in finite time)

**Pull request merged**